### DISCUSSION TITLE
iCub version

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/community#625] [Q&A] [answered]
Hi, 

how can i know iCub's version
i check the https://robot-bazaar.iit.it/robots/iCubShanghai01  it says 2.7,
can i confirm it from software or some firmware info?

**Comment by jimingda** [comment]
another question is that how can i check the electronics-wiring version since there are  4 version under v2.7
3.6.0  3.6.1  3.6.2  3.6.3.

https://github.com/icub-tech-iit/electronics-wiring-public/blob/master/icub2.7/pdf

**Comment by pattacini** [comment, marked as answer]
Hi @jimingda 

> how can i know iCub's version i check the https://robot-bazaar.iit.it/robots/iCubShanghai01 it says 2.7, can i confirm it from software or some firmware info?

As of now, there's no standard way to check the HW version of the robot from the FW and/or SW.

> another question is that how can i check the electronics-wiring version since there are 4 version under v2.7 3.6.0 3.6.1 3.6.2 3.6.3.
> 
> https://github.com/icub-tech-iit/electronics-wiring-public/blob/master/icub2.7/pdf

You ought to stick to[^1]:
- https://github.com/icub-tech-iit/electronics-wiring-public/blob/master/icub2.7/pdf/iCub2.7_E3.6.1_14839_Harness.pdf
- https://github.com/icub-tech-iit/electronics-wiring-public/blob/master/icub2.7/pdf/iCub2.7_E3.6.2_14838_Logic.pdf

[^1]: Thanks @Fabrizio69 for providing the correct revisions.
